src/init.rs

```diff
@@ -24,30 +33,254 @@ description = ""
+                    source: PathBuf::from("entry.wasm"),
+                    abi: Abi::default(),
+                    interfaces: None,
+                }
+            } else {
+                Module {
+                    name: "".to_owned(),
+                    source: PathBuf::from("none"),
+                    abi: Abi::default(),
+                    interfaces: None,
+                }
+            };
+            module.source = ask_until_valid(
+                " - Source (path)",
+                Some(module.source.to_str().unwrap().to_owned()),
+                validate_wasm_source,
+            )?;
+            if module.source.to_str().unwrap() == "none" {
+                break;
+            }
+            // Let's try to guess the name based on the file path
+            let default_module_name = Path::new(&module.source)
+                .file_stem()
+                .unwrap()
+                .to_str()
+                .unwrap()
+                .to_owned();
+            module.name = ask_until_valid(
+                " - Name",
+                Some(default_module_name.clone()),
+                util::validate_name,
+            )?;
+            let (abi, interfaces): (Abi, Option<HashMap<String, String>>) = match Select::new()
+                .with_prompt(" - ABI")
+                .item("None")
+                .item("WASI")
+                .item("Emscripten")
+                .default(0)
+                .interact()?
+            {
+                1 => (
+                    Abi::Wasi,
+                    Some(
+                        [("wasi".to_owned(), WASI_LAST_VERSION.to_owned())]
+                            .iter()
+                            .cloned()
+                            .collect(),
+                    ),
+                ),
+                2 => (Abi::Emscripten, None),
+                0 | _ => (Abi::None, None),
+            };
+            module.abi = abi;
+            module.interfaces = interfaces;
+            // We ask for commands if it has an Abi
+            if module.abi == Abi::Wasi || module.abi == Abi::Emscripten {
+                let commands = ask_until_valid(
+                    " - Commmands (space separated)",
+                    Some(default_module_name.clone()),
+                    validate_commands,
+                )?;
+                if !commands.is_empty() {
+                    all_commands.push(Command {
+                        name: commands,
+                        module: module.name.clone(),
+                        main_args: None,
+                        package: None,
+                    });
+                }
+            }
+            all_modules.push(module);
+        }
+        manifest.module = if all_modules.is_empty() {
+            None
+        } else {
+            Some(all_modules)
+        };
+        manifest.command = if all_commands.is_empty() {
+            None
+        } else {
+            Some(all_commands)
```

Comment:
It might be useful to use something like https://github.com/dac-gmbh/vec1 in the future to make the types very explicit.  It's fine to not do it in this PR though